addons/point_of_sale/models/account_journal.py

```diff
@@ -27,3 +28,55 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
     def onchange_type(self):
         if self.type not in ['bank', 'cash']:
             self.journal_user = False
+
+    @api.multi
+    def write(self, vals):
+        """ Prevent to archive journals that are still used in an opened session """
+        if not vals.get('active', True):
+            config_ids = self.env['pos.config'].search([
+                '|',
+                    ('journal_id', 'in', self.ids),
+                    ('journal_ids', 'in', self.ids)
+            ]).ids
+            if config_ids:
+                running_sessions = self.env['pos.session'].search([
+                    ('state', '!=', 'closed'),
+                    ('config_id', 'in', config_ids)
```

Comment:
the search for config above is not needed and can be done with a more complete domain here to search in one sweep